Fix issues with named data map load_into

JacobSzwejbka opened the pull request:
Summary: Fix a couple bugs in load_into and add tests

Differential Revision: D70186266

lucylq commented on extension/flat_tensor/flat_tensor_data_map.cpp:
```diff
@@ -141,7 +141,7 @@ ET_NODISCARD Result<FreeableBuffer> FlatTensorDataMap::get_data(
 }
 
-ET_NODISCARD Result<size_t> FlatTensorDataMap::load_data_into(
+ET_NODISCARD Error FlatTensorDataMap::load_data_into(
```
thanks for updating this 🙏

JacobSzwejbka committed:
Fix issues with named data map load_into (#8686)

Summary:

Fix a couple bugs in load_into and add tests

Reviewed By: lucylq

Differential Revision: D70186266
